example/__tests__/videoroomContext.test.ts

```diff
@@ -108,4 +129,68 @@ describe('Videoroom context', () => {
     expect(useAudioTrackMetadataMock).toBeCalledTimes(1);
     expect(result.current.isMicrophoneOn).toBe(false);
   });
+
+  test('Connecting to room', async () => {
+    const { result } = renderHook(() => useVideoroomState(), {
+      wrapper: VideoroomContextProvider,
+    });
+
+    act(() => {
+      result.current.setRoomName('test room     ');
+      result.current.setUsername('test user   ');
+    });
+    await act(async () => {
+      await result.current.connectAndJoinRoom();
+    });
+
+    expect(result.current.roomName).toBe('test room');
+    expect(result.current.username).toBe('test user');
+    expect(setExtra).toBeCalledTimes(2);
```

Comment:
can we check if we set correct values here?

Reply:
Done.